substrate/frame/bags-list/src/lib.rs

```diff
@@ -410,6 +405,131 @@ pub mod pallet {
 		fn try_state(_: BlockNumberFor<T>) -> Result<(), TryRuntimeError> {
 			<Self as SortedListProvider<T::AccountId>>::try_state()
 		}
+
+		/// Called during the idle phase of block execution.
+		/// Automatically performs a limited number of `rebag` operations each block,
+		/// incrementally correcting the position of accounts within the bags-list.
+		///
+		/// Guarantees processing as many nodes as possible without failing on errors.
+		/// It stores a persistent cursor to continue across blocks.
+		fn on_idle(_n: BlockNumberFor<T>, limit: Weight) -> Weight {
+			let mut meter = WeightMeter::with_limit(limit);
+			if meter.try_consume(T::WeightInfo::on_idle()).is_err() {
+				log!(debug, "Not enough Weight for on_idle. Skipping rebugging.");
+				return Weight::zero();
+			}
+
+			let rebag_budget = T::AutoRebagPerBlock::get();
+			let total_nodes = ListNodes::<T, I>::count();
```

Comment:
nit: you can fast exit if `rebag_budget` is 0 without reading the other two additional storages.